### PR TITLE
Adding Central CIS Create unittest and fix a couple of feature related errors in the process

### PR DESCRIPTION
**Pull request by erbr-ot**
Adding basic unittest for Central CIS Create.

Fix issue re. hitting ASSERT in case of non supported feature in peripheral. 

Adding API to allow unmasking peer feature, for asymmetrical features.

Fix issue re. unmasking of peer feature support in case of non support for CIS peripheral

### CODE REVIEW

**Review by mtpr-ot — DISMISSED** [review]
LGTM

File: subsys/bluetooth/controller/ll_sw/ull_llcp_cc.c [hunk 1]
```diff
@@ -852,6 +853,26 @@ static void lp_cc_st_idle(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t ev
 	}
 }
 
+static void lp_cc_state_wait_ntf_avail(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
```
Comment (mtpr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_llcp_cc.c:856]: Same as the `lp_cc_st_wait_notify_cancel` implementation in #61536. These should be aligned.

Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_llcp_cc.c:856]: When #61536 gets merged, I'll rebase and align.

**Comment by erbr-ot** [comment]
rebased to re-trigger CI

File: tests/bluetooth/controller/mock_ctrl/src/ull_central.c [hunk 2]
```diff
@@ -42,7 +42,7 @@ int ull_central_reset(void)
 }
 
 uint16_t ull_central_iso_cis_offset_get(uint16_t cis_handle, uint32_t *cis_offset_min,
-					uint32_t *cis_offset_max)
+					uint32_t *cis_offset_max, uint16_t *conn_event_count)
```
Comment (kruithofa) [inline tests/bluetooth/controller/mock_ctrl/src/ull_central.c:45]: question only: this route is not used in unittesting, correct?

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: tests/bluetooth/controller/mock_ctrl/src/ull_central.c (hunk 2 above)
Comment (erbr-ot) [reply tests/bluetooth/controller/mock_ctrl/src/ull_central.c:45]: It is used in ull_llcp_cc.c, so yes for cis unittests it is

**Review by cvinayak — APPROVED** [review]
@kruithofa have your requests addressed?

**Review by kruithofa — APPROVED** [review]

**Pull request merged**